pyfive/h5d.py

```diff
@@ -98,7 +98,21 @@ def __init__(self, dataobject, pseudo_chunking_size_MB=4):
         # throws a flake8 wobbly for Python<3.10; match is Py3.10+ syntax
         match self.layout_class:  # noqa
             case 0:  #compact storage
-                raise NotImplementedError("Compact Storage")
+                self.data = None
```

Comment:
I think it would be more consistent with the rest of the code if we pushed the compact handling into a method called from here.